src/plugins/data_views/public/saved_objects_client_wrapper.ts

```diff
@@ -21,53 +17,111 @@ import {
+  async find(options: SavedObjectsClientCommonFindArgs) {
+    const results = await this.contentManagementClient.search<
+      DataViewCrudTypes['SearchIn'],
+      DataViewCrudTypes['SearchOut']
+    >({
+      contentTypeId: DataViewSOType,
+      query: {
+        text: options.search,
+        limit: options.perPage,
+      },
+      options: {
+        searchFields: options.searchFields,
+        fields: options.fields,
+      },
+    });
+    return results.hits;
+  }
+
+  async get(id: string) {
+    let response: DataViewCrudTypes['GetOut'];
+    try {
+      response = await this.contentManagementClient.get<
+        DataViewCrudTypes['GetIn'],
+        DataViewCrudTypes['GetOut']
+      >({
+        contentTypeId: DataViewSOType,
+        id,
+      });
+    } catch (e) {
+      if (e.body?.statusCode === 404) {
+        throw new SavedObjectNotFound('data view', id, 'management/kibana/dataViews');
+      } else {
+        throw e;
+      }
+    }
+
+    if (response.meta.outcome === 'conflict') {
+      throw new DataViewSavedObjectConflictError(id);
+    }
+
+    return response.item;
   }
 
-  async get<T = unknown>(type: string, id: string) {
-    const response = await this.savedObjectClient.resolve<T>(type, id);
+  async getSavedSearch(id: string) {
+    const response = await this.savedObjectClient.resolve('search', id);
+
     if (response.outcome === 'conflict') {
       throw new DataViewSavedObjectConflictError(id);
     }
-    return simpleSavedObjectToSavedObject<T>(response.saved_object);
+    return response.saved_object;
   }
 
   async update(
-    type: string,
     id: string,
     attributes: DataViewAttributes,
-    options: SavedObjectsUpdateOptions<unknown>
+    options: DataViewCrudTypes['UpdateOptions']
   ) {
-    const response = await this.savedObjectClient.update(type, id, attributes, options);
-    return simpleSavedObjectToSavedObject(response);
+    const response = await this.contentManagementClient.update<
+      DataViewCrudTypes['UpdateIn'],
+      DataViewCrudTypes['UpdateOut']
+    >({
+      contentTypeId: DataViewSOType,
+      id,
+      data: attributes,
+      options,
+    });
+
+    return response.item as SavedObject<DataViewAttributes>;
   }
 
-  async create(type: string, attributes: DataViewAttributes, options?: SavedObjectsCreateOptions) {
-    const response = await this.savedObjectClient.create(type, attributes, options);
-    return simpleSavedObjectToSavedObject(response);
+  async create(attributes: DataViewAttributes, options: DataViewCrudTypes['CreateOptions']) {
+    const result = await this.contentManagementClient.create<
+      DataViewCrudTypes['CreateIn'],
+      DataViewCrudTypes['CreateOut']
+    >({
+      contentTypeId: DataViewSOType,
+      data: attributes,
+      options,
+    });
+
+    return result.item;
   }
 
-  delete(type: string, id: string) {
-    return this.savedObjectClient.delete(type, id, { force: true });
```

Comment:
Just played around with this: It looks like this change (not including `force: true`) prevents us from deleting a data view that is multiple spaces. This might be something we need from the content management folks.
<img width="1172" alt="image" src="https://github.com/elastic/kibana/assets/1178348/0edbc5ae-236f-4522-bb1c-c1e41e0b2d87">

Reply:
Nice catch!